someip2val/src/lib/broker_feeder/collector_client.cc

```diff
@@ -11,55 +11,81 @@
 * SPDX-License-Identifier: Apache-2.0
 ********************************************************************************/
 
+#include <sstream>
+
+#include "simple_log.h"
 #include "collector_client.h"
 
+/*** LOG helpers */
+LOGGER_STATIC_INIT("# CollectorClient::");
 
 namespace sdv {
 namespace broker_feeder {
 
-static std::string getEnvVar(const std::string& name, const std::string& defaultValue = {}) {
-    char* value = std::getenv(name.c_str());
-    return value != nullptr ? std::string(value) : std::string(defaultValue);
-}
 
-static GrpcMetadata getGrpcMetadata() {
-    GrpcMetadata grpc_metadata;
-    std::string dapr_app_id = getEnvVar("VEHICLEDATABROKER_DAPR_APP_ID");
-    if (!dapr_app_id.empty()) {
-        grpc_metadata["dapr-app-id"] = dapr_app_id;
-        std::cout << "setting dapr-app-id: " << dapr_app_id << std::endl;
-
-    }
-    return grpc_metadata;
+std::string getEnvVar(const std::string& name, const std::string& defaultValue) {
+    char* value = std::getenv(name.c_str());
+    return value != nullptr ? std::string(value) : defaultValue;
 }
 
-
-std::shared_ptr<CollectorClient> CollectorClient::createInstance(const std::string &broker_addr) {
-    return std::make_shared<CollectorClient>(broker_addr);
+std::shared_ptr<CollectorClient> CollectorClient::createInstance(std::string broker_addr, std::string auth_token) {
+    return std::make_shared<CollectorClient>(broker_addr, auth_token);
 }
 
-CollectorClient::CollectorClient(std::string broker_addr)
+CollectorClient::CollectorClient(std::string broker_addr, std::string auth_token)
     : broker_addr_(broker_addr)
+    , auth_token_(auth_token)
     , connected_(false)
 {
+    // set log level from env "KUKSA_DEBUG", defaults to info
+    LOGGER_SET_LEVEL_ENV("KUKSA_DEBUG", LEVEL_INF);
+
     changeToDaprPortIfSet(broker_addr);
     metadata_ = getGrpcMetadata();
     channel_ = grpc::CreateChannel(broker_addr, grpc::InsecureChannelCredentials());
     stub_ = sdv::databroker::v1::Collector::NewStub(channel_);
+    kuksa_stub_ = kuksa::val::v1::VAL::NewStub(channel_);
```

Comment:
How is the design? Do we use both APIs (sdv::databroker::v1 and kuksa::val::v1) in parallel? What is the benefit of that, or why is it needed?

Reply:
kuksa:: proto was introduced just for the target actuator subscribe as it didn't get merged in sdv:: proto.

sdv:: interfaces support streaming updates, that is lacking in new kuksa:: proto.
But currently we just update datapoints, not stream them anyway.